lib/i18n/backend/key_value.rb

```diff
@@ -132,20 +145,14 @@ def has_key?(key)
         def [](key)
           unless @subtree && value = @subtree[key]
             value = @store["#{@master_key}.#{key}"]
-            (@subtree ||= {})[key] = JSON.decode(value) if value
+            if value
+              value = JSON.decode(value)
+              (@subtree ||= {})[key] = value
+            end
           end
           value
         end
 
-        def is_a?(klass)
```

Comment:
I think the removal of these methods is not connected with this PR. Could you please add them back? I am concerned that it might lead to another regression.